compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.cpp

```diff
@@ -93,6 +90,70 @@ TileOp CoreOp::getTileOp() {
 // AMDAIE_DmaCpyNdBaseOp
 //===----------------------------------------------------------------------===//
 
+namespace {
+// Simplified from upstream MLIR's foldDynamicIndexList:
+LogicalResult foldMixed(SmallVectorImpl<OpFoldResult> &ofrs) {
+  bool valuesChanged = false;
+  for (OpFoldResult &ofr : ofrs) {
+    if (ofr.is<Attribute>()) continue;
+    Attribute attr;
+    if (matchPattern(ofr.get<Value>(), m_Constant(&attr))) {
+      ofr = attr;
+      valuesChanged = true;
+    }
+  }
+  return success(valuesChanged);
+}
+
+template <typename OpType, typename ReplacementBuilder>
+// Based on upstream MLIR's
+// OpWithOffsetSizesAndStridesConstantArgumentFolder
```

Comment:
```suggestion
/// Based on upstream MLIR's
/// OpWithOffsetSizesAndStridesConstantArgumentFolder
```
Nit, but IREE uses /// for function/class comments.